app/helpers/base_forms_helper.rb

```diff
@@ -0,0 +1,40 @@
+module BaseFormsHelper
+  # The standard behaviour for loading a form is to check whether the requested form matches the workflow_state for
+  # the transient registration, and redirect to the saved workflow_state if it doesn't.
+  # But if the workflow state is 'flexible' (listed below), then we skip this check and load the requested form.
+  # This means users can still navigate by using the browser back button and reload forms which don't match the
+  # saved workflow_state. We then update the workflow_state to match their request, rather than the other way around.
+  # These are generally forms after 'renewal_start_form' but before 'declaration_form'.
+  def flexible_states
```

Comment:
Apologies, if I had a bit more time I would have coded up an example so I just have my quick thoughts. My thinking is rather than have a separate list we check the state against, could we not just ask the form/controller itself?

I'm conscious the form and the state are two separate things, even though they may have the same name, and I probably need you to walk me through the flow so I understand it better.

But rather than leave you hanging (as I'll be in probably around lunchtime tomorrow) I'd blurt out my current thoughts so you have a chance to spot where I've missed the point ahead of time 😁!

So I'm looking at `set_up_form` in the forms_controller and wondering could we just make a `form_class.flexible?` call, with `flexible?` being a class method. Or something similar on the controller class? That way when we look at the code for the form/controller we can see there whether we are dealing with flexible form or not, plus we don't need to remember to keep this separate list up to date as and when we make changes to the journey.

Anyway, just my rambles. Other than that nothing else I could think to come back with.

Reply:
That makes sense to me and sounds like a much better approach! I'll have a bash at it.

Reply:
Had a go at implementing this – what do you think?